cli/command/secret/create.go

```diff
@@ -43,9 +46,23 @@ func runSecretCreate(dockerCli *command.DockerCli, options createOptions) error
 	client := dockerCli.Client()
 	ctx := context.Background()
 
-	secretData, err := ioutil.ReadAll(os.Stdin)
+	if options.file == "" {
+		return fmt.Errorf("Please specify either a file name or STDIN ('-')  with --file")
```

Comment:
Thanks @vdemeester. That is definitely not easy to spot this place 😅 . The PR has been updated, please take a look.